NDTensors/src/lib/TypeParameterAccessors/src/parameters.jl

```diff
@@ -63,3 +59,16 @@ parameter(x) = parameter(typeof(x))
 Gets the number of parameters for the Type `type`.
 """
 nparameters(type::Type) = length(parameters(type))
+
+function get_parameter(type::Type, pos)
+  param = parameter(type, pos)
+  return _get_parameter(SpecifiedParameter(param), param)
+end
+
+function _get_parameter(::SpecifiedParameter{true}, parameter)
+  return parameter
+end
+
+function _get_parameter(::SpecifiedParameter{false}, parameter)
+  return error("The requested type parameter isn't specified.")
+end
```

Comment:
I'm not sure we really need the `SpecifiedParameter` trait, I found it was sufficient for type stability just to do a runtime check of `is_parameter_specified`, see https://gist.github.com/mtfishman/c3859d89f0d0c710f5a4cb11a782ae0c, and would be a lot simpler.